### ISSUE TITLE
No module available ExAws.S3

### ISSUE THREAD

**Issue by david4shure** [ex-aws/ex_aws#549]
* Do not use the issues tracker for help or support (try Elixir Forum, Slack, IRC, etc.)
* Questions about how to contribute are fine.

### Environment

* Elixir & Erlang versions (elixir --version): Elixir 1.6.4 (compiled with OTP 19)
* ExAws version `mix deps |grep ex_aws` 
 ex_aws 2.0.2 (Hex package) (mix)
  locked at 2.0.2 (ex_aws) 8df2f96f
* HTTP client version. IE for hackney do `mix deps | grep hackney`
* hackney 1.11.0 (Hex package) (rebar3)
  locked at 1.11.0 (hackney) 4951ee01
### Current behavior
I get the following stacktrace when I reference ExAws in production

```(UndefinedFunctionError) function ExAws.S3.Upload.stream_file/1 is undefined (module ExAws.S3.Upload is not available)```

In fact, when I go to my production console, ExAws isn't even available. When I Type `Ex` then hit tab for autocomplete, it autocompletes to `Exception` not `ExAws`. 

The odd part is that I have no trouble with this code in my local development envirvonment, but when I push this out to production it can't seem to find that module.

Include code samples, errors and stacktraces if appropriate.

### Expected behavior
ExAws should be defined.

**Comment by david4shure** [comment]
Yeah I had ExAws is my `mix.exs` file. Turns out the issue was that I didn't include `ex_aws_s3` in my applications list, even though I had both `ex_aws` and `ex_aws_s3` in my deps list.

**Comment by benwilson512** [comment]
If you're on a version of Elixir >= 1.4 you no longer need to maintain the applications list FYI.

**Comment by jpinnix** [comment]
Thanks, @benwilson512. I wasn't aware of the changes in ExAWS.
